engine/worker_task_queue.go

```diff
@@ -164,7 +165,24 @@ func (w *workerTaskQueueImpl) pollAndDispatchAndPrepareNext() {
 
 func (w *workerTaskQueueImpl) commitCompletedPages(ctx context.Context) error {
 	if len(w.completedPages) > 0 {
-		// TODO optimize by combining all pages into single query (as long as it won't exceed certain limit)
+		// merge pages, e.g.,
+		// [1, 2], [3, 4], [7, 8] -> [1, 4], [7, 8]
+		sort.Slice(w.completedPages, func(i, j int) bool {
+			return w.completedPages[i].minTaskSequence < w.completedPages[j].minTaskSequence
+		})
+		var pages []*workerTaskPage
+		for _, page := range w.completedPages {
+			if len(pages) == 0 || pages[len(pages)-1].maxTaskSequence+1 < page.minTaskSequence {
+				pages = append(pages, page)
+			} else {
+				if pages[len(pages)-1].maxTaskSequence < page.maxTaskSequence {
+					pages[len(pages)-1].maxTaskSequence = page.maxTaskSequence
+				}
+			}
+		}
+
+		w.completedPages = pages
+
```

Comment:
In addition to merge interval, I was thinking that the deleting persistnce API can take a list of intervals to delete them in a single query. But you can also do it separately.

Reply:
Oh I forgot to do the sql-side. Later I did some investigations but found it cumbersome to do so. One possible solutions is to generate all task_sequence from the intervals and let the sql `delete from xxx in (..)`. But I think this is not ideal way.